### DISCUSSION TITLE
False positive for CVE-2023-1108 undertow-core

### DISCUSSION THREAD

**Discussion by bvahdat** [aquasecurity/trivy#5221] [False Detection]
### IDs

 CVE-2023-1108

### Description

Scanning a custom Docker image finds a vulnerability by the maven artifact `io.undertow:undertow-core:2.2.24.Final`

```
┌───────────────────────────────────────────────────┬──────────────────┬──────────┬───────────────────┬──────────────────────────────────┬────────────────────────────────────────────────────────────┐
│                      Library                      │  Vulnerability   │ Severity │ Installed Version │          Fixed Version           │                           Title                            │
├───────────────────────────────────────────────────┼──────────────────┼──────────┼───────────────────┼──────────────────────────────────┼────────────────────────────────────────────────────────────┤
│ io.undertow:undertow-core (my-app.jar)            │ CVE-2023-1108    │ HIGH     │ 2.2.24.Final      │ 2.3.5.Final                      │ Infinite loop in SslConduit during close                   │
│                                                   │                  │          │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-1108                  │
├───────────────────────────────────────────────────┼──────────────────┼──────────┼───────────────────┼──────────────────────────────────┼────────────────────────────────────────────────────────────┤
```

However starting from undertow `2.2.24.Final` CVE-2023-1108 is already fixed and has been tracked through:

https://issues.redhat.com/browse/UNDERTOW-2239

This can be verified through [this GitHub blame link](https://github.com/undertow-io/undertow/blame/2.2.24.Final/core/src/main/java/io/undertow/protocols/ssl/SslConduit.java#L1002-L1003) on the `2.2.24.Final` tagged codebase.

More details available by the following links:

- https://bugzilla.redhat.com/show_bug.cgi?id=2174246#c0
- https://github.com/undertow-io/undertow/pull/1457
- https://github.com/jeremylong/DependencyCheck/issues/5713#issuecomment-1542337243

### Reproduction Steps

```bash
1.
2.
3.
...
```

### Target

Filesystem

### Scanner

License

### Debug Output

```bash
.
```

### Version

```bash
v0.42.0
```

### Checklist

- [ ] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [ ] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by DmitriyLewen** [comment]
Hello @bvahdat 
Thanks for your report!

Looks like GitHub advisory database contains mistake:
<img width="799" alt="изображение" src="https://github.com/aquasecurity/trivy/assets/91113035/86fdc7b1-f418-4434-aa8a-ab29d24851ef">

Can you create an improvement for this advisory? (https://github.com/advisories/GHSA-m4mm-pg93-fv78/improve)

Regards, Dmitriy

**Reply by knqyf263** [reply]
I'll close this discussion as it is not our issue.
